packages/react-native/ReactCommon/react/renderer/core/RawValue.h

```diff
@@ -54,38 +55,52 @@ class RawValue {
   /*
    * Constructors.
    */
-  RawValue() noexcept : dynamic_(nullptr) {}
+  RawValue() noexcept : value_(folly::dynamic(nullptr)) {}
 
-  RawValue(RawValue&& other) noexcept : dynamic_(std::move(other.dynamic_)) {}
+  RawValue(RawValue&& other) noexcept : value_(std::move(other.value_)) {}
 
   RawValue& operator=(RawValue&& other) noexcept {
     if (this != &other) {
-      dynamic_ = std::move(other.dynamic_);
+      value_ = std::move(other.value_);
     }
     return *this;
   }
 
-  explicit RawValue(const folly::dynamic& dynamic) noexcept
-      : dynamic_(dynamic) {}
+  explicit RawValue(jsi::Runtime& runtime, const jsi::Value& value) noexcept
+      : value_(std::make_pair(&runtime, jsi::Value(runtime, value))) {}
+
+  explicit RawValue(jsi::Runtime& runtime, jsi::Value&& value) noexcept
+      : value_(std::make_pair(&runtime, std::move(value))) {}
+
+  explicit RawValue(const folly::dynamic& dynamic) noexcept : value_(dynamic) {}
 
   explicit RawValue(folly::dynamic&& dynamic) noexcept
-      : dynamic_(std::move(dynamic)) {}
+      : value_(std::move(dynamic)) {}
 
  private:
   friend class RawProps;
   friend class RawPropsParser;
   friend class UIManagerBinding;
 
-  /*
-   * Copy constructor and copy assignment operator would be private and only for
-   * internal use, but it's needed for user-code that does `auto val =
-   * (butter::map<std::string, RawValue>)rawVal;`
-   */
-  RawValue(const RawValue& other) noexcept : dynamic_(other.dynamic_) {}
+  RawValue(const RawValue& other) noexcept {
+    if (std::holds_alternative<folly::dynamic>(other.value_)) {
+      folly::dynamic dynamic = std::get<folly::dynamic>(other.value_);
+      value_ = dynamic;
+    } else {
+      const auto& [runtime, value] = std::get<JsiValuePair>(other.value_);
+      value_ = std::make_pair(runtime, jsi::Value(*runtime, value));
+    }
+  }
 
   RawValue& operator=(const RawValue& other) noexcept {
     if (this != &other) {
-      dynamic_ = other.dynamic_;
+      if (std::holds_alternative<folly::dynamic>(other.value_)) {
+        folly::dynamic dynamic = std::get<folly::dynamic>(other.value_);
+        value_ = dynamic;
+      } else {
+        const auto& [runtime, value] = std::get<JsiValuePair>(other.value_);
+        value_ = std::make_pair(runtime, jsi::Value(*runtime, value));
+      }
```

Comment:
Thats because I think the standard variant copy constructor is implicitly deleted because `jsi::Value` isn't copy constructible

Reply:
```
 note: copy assignment operator of 'variant<folly::dynamic, std::pair<facebook::jsi::Runtime *, facebook::jsi::Value>>' is implicitly deleted because base class '__sfinae_assign_base<__all<(is_copy_constructible_v<dynamic> && is_copy_assignable_v<dynamic>), (is_copy_constructible_v<pair<Runtime *, Value>> && is_copy_assignable_v<pair<Runtime *, Value>>)>::value, __all<(is_move_constructible_v<dynamic> && is_move_assignable_v<dynamic>), (is_move_constructible_v<pair<Runtime *, Value>> && is_move_assignable_v<pair<Runtime *, Value>>)>::value>' has a deleted copy assignment operator
```